knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md

```diff
@@ -11,9 +11,9 @@
       3. Lastly check connection with sql*plus. This will confirm username/password and wallet are valid. Template for command:
       `sqlplus <user_name>/<password>@<protocol>://<host>:<port>/<service_name>?wallet_location="<unziped_wallet_location>"`
 2. Incorrect connection identifier (specified as `database_name` in credentials dictionary in `autonomous_database.ipynb` notebook).
-   - If this is ADW then when updating credentials, for database name we want to be using the [connection identifier](https://docs.oracle.com/cd/E11882_01/network.112/e41945/glossary.htm#BGBBGCEG) which will usually end in _high, _medium or _low. All three allow you to connect to the same database but they give different levels of number of concurrent SQL statements allowed to be run along with different CPU and IO resources available for each.
+   - If this is Autonomous Database (ADB) which is the ADW and ATP databases, then when updating credentials, for database name we want to be using the [connection identifier](https://docs.oracle.com/cd/E11882_01/network.112/e41945/glossary.htm#BGBBGCEG) which will usually end in _high, _medium or _low. All three allow you to connect to the same database but they give different levels of number of concurrent SQL statements allowed to be run along with different CPU and IO resources available for each.
```

Comment:
Change:
If this is Autonomous Database (ADB) which is the ADW and ATP databases, then when updating credentials, for database name we want to be using the [connection identifier](https://docs.oracle.com/cd/E11882_01/network.112/e41945/glossary.htm#BGBBGCEG) which will usually end in _high, _medium or _low. All three allow you to connect to the same database but they give different levels of number of concurrent SQL statements allowed to be run along with different CPU and IO resources available for each.

To:
For most databases, the `database_name` variable is the name of the database. However, for any of the autonomous databases (ADB) you will need to use the  [connection identifier](https://docs.oracle.com/cd/E11882_01/network.112/e41945/glossary.htm#BGBBGCEG). Generally, this is the name of the database followed by `_high`, `_medium` or `_low`. However, your database may have non-standard connection identifier names.

Reply:
Updated!